ouroboros-network-framework/src/Ouroboros/Network/Socket.hs

```diff
@@ -351,14 +358,18 @@ connectToNode' sn handshakeCodec handshakeTimeLimits versionDataCodec NetworkCon
              traceWith muxTracer $ Mx.MuxTraceHandshakeClientError err (diffTime ts_end ts_start)
              throwIO err
 
-         Right (app, _versionNumber, _agreedOptions) -> do
+         Right (HandshakeNegotiationResult (app, _versionNumber, _agreedOptions)) -> do
              traceWith muxTracer $ Mx.MuxTraceHandshakeClientEnd (diffTime ts_end ts_start)
              bearer <- Snocket.toBearer sn sduTimeout muxTracer sd
              Mx.muxStart
                muxTracer
                (toApplication connectionId (continueForever (Proxy :: Proxy IO)) app)
                bearer
 
+         Right (HandshakeQueryResult _vMap) -> do
+             traceWith muxTracer $ Mx.MuxTraceHandshakeClientEnd (diffTime ts_end ts_start)
+             throwIO (QueryNotSupportedInThisVersion @vNumber)
```

Comment:
I think the name of the error is confusing. It seems we want to throw here, becuase we don't want to support query using `connectToNode' not because of a version number.  Isn't it?

Maybe we should just call it `QueryNotSupported` or `UnexpectedQueryError`?